### PR TITLE
ekf2: filter flow vel (used for flow velocity reset)

### PR DESCRIPTION
**Pull request by dagar**
- individual flow samples can be quite erratic

### CODE REVIEW

**Comment by AlexKlimaj** [comment]
https://github.com/AlexKlimaj/PX4-Autopilot/commits/daniel-flow-prs-2/

EKF2_OF_CTRL disabled. 
https://review.px4.io/plot_app?log=4db56bd8-8a2a-4e01-992d-4d8b71649004
https://review.px4.io/plot_app?log=cb3a2f9a-4a8a-4419-92ad-ac1a7089b4be

EKF2_OF_CTRL enabled.
https://review.px4.io/plot_app?log=bbee961c-7e84-4865-8be2-557c66e62867
https://review.px4.io/plot_app?log=0609e26d-fc60-4571-9d2f-c77b2d55b730

**Comment by dagar** [comment]
I think this also makes it a bit easier to review `estimator_optical_flow_vel` when logged at the default low rate.

image

Arguably it's worse if you want to review every raw sample full rate in `estimator_optical_flow_vel`...

**Comment by AlexKlimaj** [comment]
EKF2_OF_CTRL disabled.
https://review.px4.io/plot_app?log=5b9fe245-ebd4-428b-9de6-536f70ccfd58

EKF2_OF_CTRL enabled.
https://review.px4.io/plot_app?log=7eb06d22-0805-498b-936a-bf4f9770f23c

https://github.com/AlexKlimaj/PX4-Autopilot/commits/alex-test-7-19-2024/

**Review by AlexKlimaj — DISMISSED** [review]

File: src/modules/ekf2/EKF/ekf.h [hunk 1]
```diff
@@ -121,8 +121,8 @@ class Ekf final : public EstimatorInterface
 #if defined(CONFIG_EKF2_OPTICAL_FLOW)
 	const auto &aid_src_optical_flow() const { return _aid_src_optical_flow; }
 
-	const Vector2f &getFlowVelBody() const { return _flow_vel_body; }
-	const Vector2f &getFlowVelNE() const { return _flow_vel_ne; }
+	const Vector2f &getFlowVelBody() const { return _flow_vel_body.getState(); }
```
Comment (bresch) [inline src/modules/ekf2/EKF/ekf.h:124]: it would be nice to still log the "raw" values

**Review by bresch — CHANGES_REQUESTED** [review]

File: src/modules/ekf2/EKF/ekf.h (hunk 1 above)
Comment (dagar) [reply src/modules/ekf2/EKF/ekf.h:124]: updated to keep both

**Comment by dagar** [comment]
Rebased to resolve conflict after https://github.com/PX4/PX4-Autopilot/pull/23618.

**Review by bresch — APPROVED** [review]

**Pull request merged**